modified_cdc_protocol/operation_types/setup_pcr_plate/protocol.rb

```diff
@@ -0,0 +1,344 @@
+  # @param responses [Array<strings>]
+  def associate_lot_numbers(stripwell_list, responses)
+    stripwell_list.each do |strip_well|
+      lot_number = responses.get_response("#{LOT_NUM_KEY}#{strip_well.id}")
+      strip_well.associate(LOT_NUM_KEY, lot_number)
+    end
+  end
+
+  #===== sort compositions =====#
+
+  # Sorts the compositions into groups based on their name
+  # needed to make sure that the correct primer/probe set is added in
+  # the right row
+  #
+  # @param operation [OperationList]
+  def sort_compositions(operations: operations)
+    operations.each { |op| sort_rp_n1_n2_compositions(operation: op) }
+  end
+  
+  # Sorts the compositions into groups based on their name
+  # needed to make sure that the correct primer/probe set is added in
+  # the right row
+  #
+  # @param operation [Array<Compositions>]
+  def sort_rp_n1_n2_compositions(operation:)
+    compositions = operation.temporary[:compositions]
+    rp = []
+    n1 = []
+    n2 = []
+    compositions.each_with_index do |composition, idx|
+      name = composition.primer_probe_mix.item.sample.name.downcase
+      case
+      when name.include?('rp')
+        rp.push(composition)
+      when name.include?('n1')
+        n1.push(composition)
+      when name.include?('n2')
+        n2.push(composition)
+      end
+    end
+    operation.temporary[RP_COMPOSITION_KEY] = rp
+    operation.temporary[N1_COMPOSITION_KEY] = n1
+    operation.temporary[N2_COMPOSITION_KEY] = n2
+  end
+
+
+
+  #====== prepare materials ======#
+
+  # Prepare workspace and materials
+  #
+  # @todo Make this handle master mix or enzyme with separate
+  #   buffer dynamically
+  # @param operations [OperationList]
+  # @return [void]
+  def prepare_materials(operations:)
+    show_prepare_workspace
+    build_stripwell_master_mix_compositions(operations: operations)
+  end
+  
+
+  #====== assemble primer_probe plates ======#
+
+  # Assembles the primer probe stripwells in the 96 well rack
+  #
+  # @param operations [Array<Operations>] 
+  def assemble_primer_probe_plates(operations:)
+    operations.each { |op| assemble_primer_probe_plate(operation: op) }
+  end
+
+  # Assembles the primer probe stripwells in the 96 well rack
+  #
+  # @param operation [Operation]
+  def assemble_primer_probe_plate(operation:)
+    method = operation.temporary[:options][:layout_method]
+    group_size = operation.temporary[:options][:group_size]
+    program_name = operation.temporary[:options][:program_name]
+
+    output_collection = operation.output(PLATE).collection
+    output_collection.associate(PRIMER_GROUP_SIZE_KEY, group_size)
+    output_collection.associate(COMPOSITION_NAME_KEY, program_name)
+    output_collection.associate(PRIMER_METHOD_KEY, method)
+
+    microtiter_plate = MicrotiterPlateFactory.build(
+      collection: output_collection,
+      group_size: group_size,
+      method: method
+    )
+    get_composition_keys(method).each do |composition_key|
```

Comment:
Yeah, see, now you need a method that knows about all this other stuff to know how to get your compositions, when actually you just want to iterate over them.